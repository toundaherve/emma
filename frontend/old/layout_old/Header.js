import React from "react";
import styled from "styled-components";
import Logo from "../components/Logo";

import theme from "../theme";

const Header = styled.div`
  height: 4.5rem;
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: #ffffff;
  border: 1px solid #cccccc;
  padding: 0 ${theme.spacing[7]};
`;

export default () => {
  return (
    <Header>
      <Logo />
    </Header>
  );
};
