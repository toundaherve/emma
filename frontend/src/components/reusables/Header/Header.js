import React from "react";
import { Link } from "react-router-dom";

const Header = () => {
  return (
    <header>
      <div className="user-account-bar logged-out"></div>
      <div id="header-container" className="container navigation">
        <Link
          to="/"
          id="chili-logo"
          className="logo"
          title="Chili's Grill &amp; Bar - Local Restaurants Near Me | Chili's"
        >
          Chili's Grill &amp; Bar - Local Restaurants Near Me | Chili's
        </Link>
        <Link
          to="/cart"
          id="header-cart"
          className="cart-btn js-cart-btn"
          data-cart-has-items="false"
        >
          <div className="count">0</div>
          <span className="items">0 items</span>
          <img
            className="bag js-bag"
            alt="Cart"
            src="data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCA5NiA5NiI+PGRlZnM+PHN0eWxlPi5jbHMtMXtmaWxsOiM0YTRhNGE7fTwvc3R5bGU+PC9kZWZzPjx0aXRsZT5iYWc8L3RpdGxlPjxnIGlkPSJfNGE0YTRhIiBkYXRhLW5hbWU9IjRhNGE0YSI+PHBhdGggY2xhc3M9ImNscy0xIiBkPSJNNjQuNSwzM1YyMC41YTE2LjUsMTYuNSwwLDAsMC0zMywwVjMzSDExVjc3LjRBMTQuNiwxNC42LDAsMCwwLDI1LjYsOTJINzAuNEExNC42LDE0LjYsMCwwLDAsODUsNzcuNFYzM1ptLTI3LTEyLjVhMTAuNSwxMC41LDAsMCwxLDIxLDBWMzNoLTIxWiIvPjwvZz48L3N2Zz4="
          />
        </Link>
        <a href="#" className="nav-toggle js-nav-toggle">
          Menu
        </a>
        <nav className="nav-collapse animated fadeIn js-nav-collapse">
          <ul id="header-menu-list">
            <li>
              <Link to="/" id="header-order" className="header-link">
                Home
              </Link>
            </li>
            <li>
              <Link to="/menu" id="header-menu" className="header-link">
                Menu
              </Link>
            </li>

            <li>
              <Link to="/" id="header-login" className="header-link">
                Log In
              </Link>
            </li>
          </ul>
        </nav>
        <div className="mini-cart-form animated fadeIn js-mini-cart-form">
          <div className="arrow-up"></div>
          <div className="arrow-up-2"></div>
          <div className="row header-mini-cart-form">
            <div className="heading-secondary mini-cart-title">Your Order</div>
          </div>
          <div className="row">
            <div className="col-mini-cart-item mini-cart-heading">Item</div>
            <div className="col-mini-cart-qty mini-cart-heading">Qty</div>
          </div>
          <div className="mini-cart-total row">
            <div className="col6 total-label">Subtotal</div>
            <div className="col6 total-value">
              <span></span>
            </div>
            <div className="mini-cart-actions btn-container center">
              <a
                id="mini-cart-view-upsell"
                className="btn small js-view-cart-btn"
                href="/cart"
              >
                View Cart
              </a>
              <a
                id="mini-cart-add-food"
                className="btn small tertiary outline js-close-btn"
                href="/menu"
              >
                Add Food
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="location-bar text-center"
        aria-label="Restaurant location bar"
        style={{ position: "static", top: "0px" }}
      >
        <div className="container location-bar-wrapper">
          <a
            id="location-bar-select-location"
            className="location-bar-link"
            href="/locations"
          >
            <span className="location-bar-icon icon-search-location"></span>
            <span className="location-bar-text">Enter your postcode</span>
          </a>
        </div>
      </div>
    </header>
  );
};

export default Header;
