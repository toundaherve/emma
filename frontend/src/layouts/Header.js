import React from "react";
import { AppBar, Box, Grid, Typography } from "@material-ui/core";
import { LocalMall } from "@material-ui/icons";

import Container from "../components/Container";
import Logo from "../components/Logo";
import useDeviceType from "../hooks/useDeviceType";
import { useLocation } from "react-router-dom";

const Header = ({ onShoppingCartIconClick }) => {
  const isTablet = useDeviceType("tablet");
  const isMenuPage = useLocation().pathname === "/menu";
  return (
    <AppBar>
      <Container>
        <Box
          height={`${isTablet ? "80px" : "72px"}`}
          display="flex"
          alignItems="center"
          justifyContent="space-between"
        >
          <Logo />
          {isMenuPage && <ShoppingCartIcon onClick={onShoppingCartIconClick} />}
        </Box>
      </Container>
    </AppBar>
  );
};

const ShoppingCartIcon = ({ onClick }) => {
  return (
    <Box
      pt={1.5}
      pb={1.5}
      pl={2}
      pr={2}
      onClick={onClick}
      style={{ cursor: "pointer" }}
    >
      <Grid container spacing={1} alignItems="center">
        <Grid item>
          <LocalMall size="medium" />
        </Grid>

        <Grid item>
          <Typography variant="body1">2</Typography>
        </Grid>
      </Grid>
    </Box>
  );
};

export default Header;
