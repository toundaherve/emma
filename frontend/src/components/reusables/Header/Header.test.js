import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";

import Header from "./Header";

it("renders correctly", () => {
  const component = renderer.create(
    <Router>
      <Header />
    </Router>
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
