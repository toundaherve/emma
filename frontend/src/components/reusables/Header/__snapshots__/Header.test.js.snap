// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<header>
  <div
    className="user-account-bar logged-out"
  />
  <div
    className="container navigation"
    id="header-container"
  >
    <a
      className="logo"
      href="/"
      id="chili-logo"
      onClick={[Function]}
      title="Chili's Grill & Bar - Local Restaurants Near Me | Chili's"
    >
      Chili's Grill & Bar - Local Restaurants Near Me | Chili's
    </a>
    <a
      className="cart-btn js-cart-btn"
      data-cart-has-items="false"
      href="/cart"
      id="header-cart"
      onClick={[Function]}
    >
      <div
        className="count"
      >
        0
      </div>
      <span
        className="items"
      >
        0 items
      </span>
      <img
        alt="Cart"
        className="bag js-bag"
        src="data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCA5NiA5NiI+PGRlZnM+PHN0eWxlPi5jbHMtMXtmaWxsOiM0YTRhNGE7fTwvc3R5bGU+PC9kZWZzPjx0aXRsZT5iYWc8L3RpdGxlPjxnIGlkPSJfNGE0YTRhIiBkYXRhLW5hbWU9IjRhNGE0YSI+PHBhdGggY2xhc3M9ImNscy0xIiBkPSJNNjQuNSwzM1YyMC41YTE2LjUsMTYuNSwwLDAsMC0zMywwVjMzSDExVjc3LjRBMTQuNiwxNC42LDAsMCwwLDI1LjYsOTJINzAuNEExNC42LDE0LjYsMCwwLDAsODUsNzcuNFYzM1ptLTI3LTEyLjVhMTAuNSwxMC41LDAsMCwxLDIxLDBWMzNoLTIxWiIvPjwvZz48L3N2Zz4="
      />
    </a>
    <a
      className="nav-toggle js-nav-toggle"
      href="#"
    >
      Menu
    </a>
    <nav
      className="nav-collapse animated fadeIn js-nav-collapse"
    >
      <ul
        id="header-menu-list"
      >
        <li>
          <a
            className="header-link"
            href="/"
            id="header-order"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li>
          <a
            className="header-link"
            href="/menu"
            id="header-menu"
            onClick={[Function]}
          >
            Menu
          </a>
        </li>
        <li>
          <a
            className="header-link"
            href="/"
            id="header-login"
            onClick={[Function]}
          >
            Log In
          </a>
        </li>
      </ul>
    </nav>
    <div
      className="mini-cart-form animated fadeIn js-mini-cart-form"
    >
      <div
        className="arrow-up"
      />
      <div
        className="arrow-up-2"
      />
      <div
        className="row header-mini-cart-form"
      >
        <div
          className="heading-secondary mini-cart-title"
        >
          Your Order
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-mini-cart-item mini-cart-heading"
        >
          Item
        </div>
        <div
          className="col-mini-cart-qty mini-cart-heading"
        >
          Qty
        </div>
      </div>
      <div
        className="mini-cart-total row"
      >
        <div
          className="col6 total-label"
        >
          Subtotal
        </div>
        <div
          className="col6 total-value"
        >
          <span />
        </div>
        <div
          className="mini-cart-actions btn-container center"
        >
          <a
            className="btn small js-view-cart-btn"
            href="/cart"
            id="mini-cart-view-upsell"
          >
            View Cart
          </a>
          <a
            className="btn small tertiary outline js-close-btn"
            href="/menu"
            id="mini-cart-add-food"
          >
            Add Food
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    aria-label="Restaurant location bar"
    className="location-bar text-center"
    style={
      Object {
        "position": "static",
        "top": "0px",
      }
    }
  >
    <div
      className="container location-bar-wrapper"
    >
      <a
        className="location-bar-link"
        href="/locations"
        id="location-bar-select-location"
      >
        <span
          className="location-bar-icon icon-search-location"
        />
        <span
          className="location-bar-text"
        >
          Enter your postcode
        </span>
      </a>
    </div>
  </div>
</header>
`;
